From 04a628c2b0c3dc9fdbdfc654152b47dfaeabd3bc Mon Sep 17 00:00:00 2001
From: vovagusev <v@vovagusev.ru>
Date: Thu, 30 May 2024 21:50:30 +0400
Subject: [PATCH 10/28] Improve error handling and logging in get_assignee_name

---
 mybot/bot.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/mybot/bot.py b/mybot/bot.py
index 9545ebe..e6c8024 100644
--- a/mybot/bot.py
+++ b/mybot/bot.py
@@ -229,11 +229,12 @@ async def add_task(update: Update, context: CallbackContext) -> None:
 
 
 async def get_assignee_name(bot, chat_id, assignee_data):
-    if assignee_data is not None and 'id' in assignee_data:  # Добавлена проверка на None
+    if assignee_data is not None and 'id' in assignee_data:
         try:
             assignee_user = await bot.get_chat_member(chat_id, assignee_data['id'])
             return assignee_user.user.full_name
         except BadRequest:
+            logging.error(f"Ошибка при получении информации о пользователе: {assignee_data}")
             return "Неизвестный пользователь"
     else:
         return "Не назначен"
-- 
2.45.1.windows.1

