From 64bd47f384c1ff88fefc3028924423940d718534 Mon Sep 17 00:00:00 2001
From: vovagusev <v@vovagusev.ru>
Date: Thu, 30 May 2024 22:45:30 +0400
Subject: [PATCH 17/28] Fix coroutine not awaited issue in done_task function

---
 mybot/bot.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mybot/bot.py b/mybot/bot.py
index 8da745d..c1afd70 100644
--- a/mybot/bot.py
+++ b/mybot/bot.py
@@ -401,7 +401,7 @@ async def done_task(update: Update, context: CallbackContext) -> None:
         chat_tasks[chat_id][task_number]['status'] = 'Завершена'
         await update_pinned_message(chat_id, context.bot)
         save_state(chat_tasks, pinned_message_id)
-        assignee_name = get_assignee_name(
+        assignee_name = await get_assignee_name(
             context.bot, chat_id, chat_tasks[chat_id][task_number].get('assignee', None)
         )
         await update.message.reply_text(f"Задача {task_number + 1} выполнена ({assignee_name}).")
-- 
2.45.1.windows.1

