From df17bec02ff252b1c9ba4ee63278596f5a42f868 Mon Sep 17 00:00:00 2001
From: vovagusev <v@vovagusev.ru>
Date: Thu, 30 May 2024 22:54:22 +0400
Subject: [PATCH 18/28] Fix assignee name retrieval in done_task function

---
 mybot/bot.py | 9 ++++-----
 1 file changed, 4 insertions(+), 5 deletions(-)

diff --git a/mybot/bot.py b/mybot/bot.py
index c1afd70..a4eaa2b 100644
--- a/mybot/bot.py
+++ b/mybot/bot.py
@@ -234,12 +234,12 @@ async def add_task(update: Update, context: CallbackContext) -> None:
 
 
 async def get_assignee_name(bot, chat_id, assignee_id):
-    if assignee_id:
+    if assignee_id is not None:
         try:
             assignee_user = await bot.get_chat_member(chat_id, assignee_id)
             return assignee_user.user.full_name
         except BadRequest:
-            logging.error(f"Ошибка при получении информации о пользователе: {assignee_id}")
+            logging.error(f"Ошибка при получении информации о пользователе с ID: {assignee_id}")
             return "Неизвестный пользователь"
     else:
         return "Не назначен"
@@ -401,9 +401,8 @@ async def done_task(update: Update, context: CallbackContext) -> None:
         chat_tasks[chat_id][task_number]['status'] = 'Завершена'
         await update_pinned_message(chat_id, context.bot)
         save_state(chat_tasks, pinned_message_id)
-        assignee_name = await get_assignee_name(
-            context.bot, chat_id, chat_tasks[chat_id][task_number].get('assignee', None)
-        )
+        assignee_data = chat_tasks[chat_id][task_number].get('assignee', {})
+        assignee_name = await get_assignee_name(context.bot, chat_id, assignee_data.get('id'))
         await update.message.reply_text(f"Задача {task_number + 1} выполнена ({assignee_name}).")
     else:
         await update.message.reply_text("Некорректный номер задачи.")
-- 
2.45.1.windows.1

