From 01dae0f46e3280e3152256ac56fdd5b927cb9c6a Mon Sep 17 00:00:00 2001
From: vovagusev <v@vovagusev.ru>
Date: Fri, 31 May 2024 02:26:40 +0400
Subject: [PATCH 26/28] Remove incorrect assignee information from task
 completion message

---
 mybot/bot.py | 10 ++++------
 1 file changed, 4 insertions(+), 6 deletions(-)

diff --git a/mybot/bot.py b/mybot/bot.py
index 5c121dc..58ebd14 100644
--- a/mybot/bot.py
+++ b/mybot/bot.py
@@ -338,13 +338,11 @@ async def done_task(update: Update, context: CallbackContext) -> None:
         chat_tasks[chat_id][task_number]['status'] = 'Завершена'
         await update_pinned_message(chat_id, context.bot)
         save_state(chat_tasks, pinned_message_id)
-        assignee = chat_tasks[chat_id][task_number].get('assignee', None)
+        assignee_data = chat_tasks[chat_id][task_number].get('assignee', None)
         assignee_name = ""
-        if assignee:
-            assignee_name = await get_assignee_name(context.bot, chat_id, assignee)
-        await update.message.reply_text(
-            f"Задача {task_number + 1} выполнена" + (f" ({assignee_name})" if assignee_name else "") + "."
-        )
+        if assignee_data and assignee_data.get('id'):
+            assignee_name = await get_assignee_name(context.bot, chat_id, assignee_data.get('id'))
+        await update.message.reply_text(f"Задача {task_number + 1} выполнена" + (f" ({assignee_name})" if assignee_name else "") + ".")
     else:
         await update.message.reply_text("Некорректный номер задачи.")
 
-- 
2.45.1.windows.1

